src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Well basically your reasoning is:  
> ... because F was the first child of C, so C should now be the first child of F.
While this is true and reasonable, I do not think, there is a use case for it.

Also what happens, if their is one layer between them? (Serious question, if you still want it to be implemented this way.)

I think just guaranteeing to add it to the end and not disturb the rest of the children is the best idea, because you want to keep the children, when swapping. Which means to me, that most likely you do not want to change the order of them by having a node inserted somewhere in the middle.